File: CustomTextInput/ViewController.m
//
//  ViewController.m
//  CustomTextInput
//
//  Created by janezhuang on 2022/7/22.
//

#import "ViewController.h"
#import "EditableCoreTextView.h"

@interface ViewController ()
@property (nonatomic) EditableCoreTextView *textView;
@end

@implementation ViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.textView = [[EditableCoreTextView alloc] initWithFrame:CGRectMake(60, 120, 200, 200)];
    self.textView.backgroundColor = UIColor.lightGrayColor;
    [self.view addSubview:self.textView];
}
@end

File: CoreTextMagazine/ViewController.swift
//
//  ViewController.swift
//  CoreTextMagazine
//
//  Created by janezhuang on 2023/5/9.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        // 1
        guard let file = Bundle.main.path(forResource: "zombies", ofType: "txt") else { return }

        do {
            let text = try String(contentsOfFile: file, encoding: .utf8)
            // 2
            let parser = MarkupParser()
            parser.parseMarkup(text)
            (view as? CTView)?.buildFrames(withAttrString: parser.attrString, andImages: parser.images)

        } catch _ {
        }
    }

}
